default/python/AI/PolicyAI.py

```diff
@@ -196,18 +232,39 @@ def _unlocked_by(name: str) -> Optional[str]:
             return diversity
         return None
 
+    _rating_functions = {
+        propaganda: lambda self: 20 + self._rate_opinion(propaganda),
+        algo_research: lambda self: self._rate_algo_research(),
+        diversity: lambda self: self._rate_diversity(),
+        artisans: lambda self: self._rate_artisans(),
+        # military policies are mostly chosen by opinion, plus some rule-of-thumb values
+        allied_repair: lambda self: self._rate_opinion(charge),  # no effect, unless we have allies...
+        charge: lambda self: 5 + self._rate_opinion(charge),  # A small bonus in battle
+        scanning: lambda self: 20 + self._rate_opinion(scanning),  # May help us detect ships and planets
+        simplicity: lambda self: 20 + self._rate_opinion(simplicity),  # Makes simple ships cheaper
+        engineering: lambda self: self._rate_engineering_corps(),
+        exploration: lambda self: 10 + self._rate_opinion(exploration),  # may give a little research, speeds up scouts
+        flanking: lambda self: 5 + self._rate_opinion(flanking),  # A small bonus in battle
+        recruitment: lambda self: 15 + self._rate_opinion(recruitment),  # cheaper troop ships
+    }
+
     @cache_for_current_turn
     def _rate_policy(self, name: str) -> float:
         """Gives a rough value of a policy."""
-        if name == propaganda:
-            return 20 + self._rate_opinion(name)
-        if name == algo_research:
-            return self._rate_algo_research()
-        if name == diversity:
-            return self._rate_diversity()
-        if name == artisans:
-            return self._rate_artisans()
-        raise (ValueError(f"_rate_policy {name} not yet supported"))
+        rating_function = self._rating_functions.get(name)
+        if assertion_fails(rating_function, f"_rate_policy({name}) not yet supported"):
+            return 0.0
+        return rating_function(self)
+
+    def _rate_costs(self, costs: float) -> float:
+        """
+        How do we rate the costs in comparison to a policies rating?
+        Generally costs / 3, but less if we have plenty of IP.
+        """
+        unreserved_ip = self._ip - self._wanted_ip
+        priority = max(self._aistate.get_priority(PriorityType.RESOURCE_INFLUENCE), 1)  # in turn 1 priority is 0
+        # For big numbers this makes 1.5 * priority an upper limit for the rating
```

Comment:
Not sure that I got this comment, could you rephrase it?

Reply:
I refactored the whole function. Using min instead of costs / max it should be easier to read and I don't need the special handling for turn 1 anymore.